diff --git a/config/schema/inline_entity_form.schema.yml b/config/schema/inline_entity_form.schema.yml
index b962119..fef9385 100644
--- a/config/schema/inline_entity_form.schema.yml
+++ b/config/schema/inline_entity_form.schema.yml
@@ -57,6 +57,9 @@ field.widget.settings.inline_entity_form_complex:
     allow_existing:
       type: boolean
       label: "Allow existing"
+    allow_edit:
+      type: boolean
+      label: "Allow edit"
     match_operator:
       type: string
       label: "Match operator"
diff --git a/src/Plugin/Field/FieldWidget/InlineEntityFormComplex.php b/src/Plugin/Field/FieldWidget/InlineEntityFormComplex.php
index a4b2c30..4fcf9b6 100644
--- a/src/Plugin/Field/FieldWidget/InlineEntityFormComplex.php
+++ b/src/Plugin/Field/FieldWidget/InlineEntityFormComplex.php
@@ -104,6 +104,7 @@ class InlineEntityFormComplex extends InlineEntityFormBase implements ContainerF
     $defaults += [
       'allow_new' => TRUE,
       'allow_existing' => FALSE,
+      'allow_edit' => TRUE,
       'match_operator' => 'CONTAINS',
       'allow_duplicate' => FALSE,
     ];
@@ -129,6 +130,11 @@ class InlineEntityFormComplex extends InlineEntityFormBase implements ContainerF
       '#title' => $this->t('Allow users to add existing @label.', ['@label' => $labels['plural']]),
       '#default_value' => $this->getSetting('allow_existing'),
     ];
+    $element['allow_edit'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Allow users to edit existing @label.', ['@label' => $labels['plural']]),
+      '#default_value' => $this->getSetting('allow_edit'),
+    ];
     $element['match_operator'] = [
       '#type' => 'select',
       '#title' => $this->t('Autocomplete matching'),
@@ -352,7 +358,7 @@ class InlineEntityFormComplex extends InlineEntityFormBase implements ContainerF
 
         // Make sure entity_access is not checked for unsaved entities.
         $entity_id = $entity->id();
-        if (empty($entity_id) || $entity->access('update')) {
+        if (empty($entity_id) || ($settings['allow_edit'] && $entity->access('update'))) {
           $row['actions']['ief_entity_edit'] = [
             '#type' => 'submit',
             '#value' => $this->t('Edit'),